src/amo/withExperiment.js

```diff
@@ -200,10 +204,12 @@ export const withExperiment = ({
         experimentsToStore[id] = variantToStore.id;
 
         if (variantToStore.id !== NOT_IN_EXPERIMENT) {
-          // send an enrollment event
+          // Send an enrollment event.
+          // Because this might be run on the server, we must provide a dispatch argument.
           _tracking.sendEvent({
```

Comment:
Can this NOT run on the server? I am thinking that we might want to be more explicit and have a `sendEventFromServer` function or something.

Reply:
I'm not sure I follow. This code could be called on both the client and the server, so the idea was to have a function that will work on both the client and the server. I believe you were the one who suggested putting the code to support this into `tracking`. Are you suggesting we detect *here* whether we are on the client or the server, and call different tracking functions depending on that? That kind of defeats the point of moving the logic into `tracking`, doesn't it?

Reply:
I am not suggesting anything, I don't like that we pass `dispatch` to `sendEvent` but I don't have any ideas to improve the situation so I guess we'll leave that to the future AMO team 😉

Reply:
I don't deny it feels a bit ugly, but it works well, and allows `tracking` to deal with the issue, rather than `withExperiment` (or anything else that requires it in the future), so maybe even the future AMO team will be okay with it. 😉